### ISSUE TITLE
"Update Sidebar Ordering"button does not work.

### ISSUE THREAD

**Issue by luoyanjie** [Coursemology/coursemology2#2321]
Under course settings > Sidebar

**Comment by allenwq** [comment]
The re-order does not work if:
1. The course has more than 2 assessment categories
2. Move any sidebar items between the 2 categories

All assessment categories have the same ordering key `assessments`. 
This should relate to #1795, should give different categories a different key.

**Comment by fonglh** [comment]
However, if the assessment categories are moved together as 1 unit, ordering still works.

Sample psql output:
```
-[ RECORD 1 ]----+-------------------------------------------------------------------------------
id               | 2
instance_id      | 0
title            | Enrollable Course
description      | example course
logo             |
published        | t
enrollable       | t
registration_key |
settings         | --- !ruby/hash:ActiveSupport::HashWithIndifferentAccess                       +
                 | sidebar: !ruby/hash:ActiveSupport::HashWithIndifferentAccess                  +
                 |   announcements: !ruby/hash:ActiveSupport::HashWithIndifferentAccess          +
                 |     weight: 3                                                                 +
                 |   assessments_submissions: !ruby/hash:ActiveSupport::HashWithIndifferentAccess+
                 |     weight: 11                                                                +
                 |   achievements: !ruby/hash:ActiveSupport::HashWithIndifferentAccess           +
                 |     weight: 5                                                                 +
                 |   discussion_topics: !ruby/hash:ActiveSupport::HashWithIndifferentAccess      +
                 |     weight: 4                                                                 +
                 |   leaderboard: !ruby/hash:ActiveSupport::HashWithIndifferentAccess            +
                 |     weight: 6                                                                 +
                 |   users: !ruby/hash:ActiveSupport::HashWithIndifferentAccess                  +
                 |     weight: 7                                                                 +
                 |   lesson_plan: !ruby/hash:ActiveSupport::HashWithIndifferentAccess            +
                 |     weight: 8                                                                 +
                 |   materials: !ruby/hash:ActiveSupport::HashWithIndifferentAccess              +
                 |     weight: 9                                                                 +
                 |   forums: !ruby/hash:ActiveSupport::HashWithIndifferentAccess                 +
                 |     weight: 12                                                                +
                 |   surveys: !ruby/hash:ActiveSupport::HashWithIndifferentAccess                +
                 |     weight: 10                                                                +
                 |   assessments: !ruby/hash:ActiveSupport::HashWithIndifferentAccess            +
                 |     weight: 2                                                                 +
```

**Comment by allenwq** [comment]
The bug should be fixed in #2547. Cannot split the category is a different issue.
